mozjs/src/conversions.rs

```diff
@@ -568,15 +571,13 @@ pub unsafe fn jsstr_to_string(cx: *mut JSContext, jsstr: NonNull<JSString>) -> S
 impl ToJSValConvertible for str {
     #[inline]
     unsafe fn to_jsval(&self, cx: *mut JSContext, mut rval: MutableHandleValue) {
-        let mut string_utf16: Vec<u16> = Vec::with_capacity(self.len());
-        string_utf16.extend(self.encode_utf16());
-        let jsstr = JS_NewUCStringCopyN(
-            cx,
-            string_utf16.as_ptr(),
-            string_utf16.len() as libc::size_t,
-        );
+        // Spidermonkey will automatically only copy latin1
+        // or similar if the given encoding can be small enough.
+        // So there is no need to distinguish between ascii only or similar.
+        let s = Utf8Chars::from(self);
+        let jsstr = JS_NewStringCopyUTF8N(cx, &*s as *const _);
```

Comment:
- for new code it is preferable to use `<ptr>.cast()` / `<ptr>.cast_mut()` instead of `as *const _` / `as *mut _`
- could you add the lint that requires unsafe blocks in unsafe fns to this function? Then perhaps move the `&*s` to another line, since that cast should be safe, right? (It's not visible from the diff right now due to missing type information / no unsafe blocks.